src/HDF5.jl

```
@@ -1926,11 +1927,11 @@ function h5d_write{T<:HDF5Scalar}(dataset_id::Hid, memtype_id::Hid, x::T)
     h5d_write(dataset_id, memtype_id, H5S_ALL, H5S_ALL, H5P_DEFAULT, tmp)
 end
 function h5d_write{S<:String}(dataset_id::Hid, memtype_id::Hid, strs::Array{S})
-    len = length(strs)
-    p = Array{Ptr{UInt8}}(size(strs))
-    for i = 1:len
-        p[i] = !isassigned(strs, i) || isempty(strs[i]) ? pointer(EMPTY_STRING) : pointer(strs[i])
+    nonnull_str = copy(strs)
+    for i = 1:length(nonnull_str)
+        isassigned(nonnull_str, i) || (nonnull_str[i] = "")
     end
+    p = Ref{Cstring}(nonnull_str)
```

Right, the last element should be `""`. Why would you ever want it to silently discard / truncate your data? :)